src/components/dialog/CloseButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react'
+import { css } from '../../../styled-system/css'
+import CloseIcon from '../icons/CloseIcon'
+
+interface CloseButtonProps {
+  onClick: () => void
+}
+
+/**
+ * Dialog close button.
+ */
+const CloseButton: React.FC<CloseButtonProps> = ({ onClick }) => {
+  return (
+    <button
+      onClick={onClick}
+      className={css({
+        width: '40px',
+        height: '40px',
+        borderRadius: '50%',
+        marginRight: '0.5rem',
+        backgroundColor: '{colors.darkgray}',
+        border: `2px solid {colors.gray50}`,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        cursor: 'pointer',
+        transition: 'background-color 0.2s ease-in-out',
```

Comment:
You can use a predefined duration from https://github.com/cybersemics/em/blob/f8df451f3e3a4b45697d4e5d8167ead3373af52c/src/durations.config.ts.